### ISSUE TITLE
Pushdown functions with OUT parameters

### ISSUE THREAD

**Issue by SaitTalhaNisanci** [citusdata/citus#5230]
With PG14, procedures can have `OUT` parameters.  We currently error out instead of pushing them down to workers. We should implement the logic to push them down.

Relevant PR that errors out: https://github.com/citusdata/citus/pull/5228

**Comment by onderkalaci** [comment]
> Relevant PR that errors out: #5228

just a note, we do not error out, but instead we do not pushdown (e.g., execute the commands in the function on the coordinator one by one)

**Comment by SaitTalhaNisanci** [comment]
Yeah, for some reason I thought not pushing down as erroring out haha Thanks for correcting
